pkg/services/ngalert/alert_definition.go

```diff
@@ -0,0 +1,71 @@
+package ngalert
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana/pkg/services/ngalert/eval"
+)
+
+// GetEvalCondition returns a condition suitable for execution
+// based on the saved AlertDefinition
+func (alertDefinition *AlertDefinition) GetEvalCondition(now time.Time) (*eval.Condition, error) {
+	// Copy Properties backend.DataQuery
+	// Turn RelativeTimeRange into a time Range
+	// Copy the AlertQuery's Model, update model in DataQuery as needed
+
+	condition := &eval.Condition{
+		RefID: alertDefinition.Condition,
+	}
+	for _, aq := range alertDefinition.Data {
+		// GEL requires it (fails with InvalidArgument desc = no orgId in gel command for refId if not set)
+		// this should be the organization with which the alert definition is created
+		// or organisation of the signed in user???
+		err := aq.setOrgID(alertDefinition.OrgId)
+		if err != nil {
+			return nil, fmt.Errorf("failed to set orgId query model %w", err)
+		}
+
+		model, err := aq.getModel()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get query model %w", err)
+		}
+
+		intervalMs, err := aq.getIntervalMs()
+		if err != nil {
+			return nil, fmt.Errorf("failed to retrieve intervalMs from the model %w", err)
+		}
+
+		maxDatapoints, err := aq.getMaxDatapoints()
+		if err != nil {
+			return nil, fmt.Errorf("failed to retrieve maxDatapoints from the model %w", err)
+		}
+		condition.QueriesAndExpressions = append(condition.QueriesAndExpressions, backend.DataQuery{
+			JSON:          model,
+			Interval:      time.Duration(intervalMs) * time.Millisecond,
+			RefID:         aq.RefID,
+			MaxDataPoints: maxDatapoints,
+			QueryType:     "",
```

Comment:
important we set this one for certain datasources I think

Reply:
I want to ask about this: this should be an additional field in AlertQuery or will be retrieved from the JSON data?